### DISCUSSION TITLE
Usage with linaria / getPageAssets

### DISCUSSION THREAD

**Discussion by 0x241F31** [vikejs/vike#688] [Discussion]
For my blog, I wanted to use [Linaria](https://github.com/callstack/linaria) with this beautiful plugin. I experimented with injecting of generated CSS directly in html head `<style>`. By default Vite inserts virtual css imports that Linaria produces into `<link>` imports. With `injectFilter` it's possible to control this, but before usage of filter, I need to get list of generated css imports, and the only way to get them I've found is to use `getPageAssets`. Are there another ways? If no, maybe it's better to not deprecate `getPageAssets`?

```ts
const assetsToSkip = new Set()
if (!viteDevServer && _isHtmlOnly) {
   const assets = await __getPageAssets()
   for (const { src, assetType } of assets) {
      if (assetType != "style") continue
      const css = await readFile(join(root, outDir, src))
      head.unshift(`<style>${css}</style>`)
      assetsToSkip.add(src)
      // changing asset.inject to false here doesn't do anything, therefore using injectFilter
   }
}
const injectFilter: (assets: InjectFilterEntry[]) => {
   if (!ctx._isHtmlOnly) return
   for (const a of assets) assetsToSkip.has(a.src) && (a.inject = false) 
},
```

**Comment by brillout** [comment]
I see. Ideally that's something vite-plugin-ssr should do by default. (With the possibility to opt-out using [`injectFilter()`](https://vite-plugin-ssr.com/injectFilter).)

Feel free to create a feature request. Although I most likely won't implement this anytime soon (there are currently a couple of higher priorities). That said, PR welcome.
